### PR TITLE
Improve IoTHub_CommunicationManager_Init and DeInit

### PR DESCRIPTION
**Pull request by Nox-MSFT**
Moved IoTHub_DeInit() call into IoTHub_CommunicationManager_DeInit() and only call if the IoTHub_Init() has been called.

### CODE REVIEW

File: src/communication_managers/iothub_communication_manager/src/iothub_communication_manager.c [hunk 1]
```diff
@@ -106,23 +111,39 @@ static time_t GetTimeSinceEpochInSeconds()
...
     ADUC_ClientHandle* handle_address,
     IOTHUB_CLIENT_DEVICE_TWIN_CALLBACK device_twin_callback,
     ADUC_COMMUNICATION_MANAGER_CLIENT_HANDLE_UPDATED_CALLBACK client_handle_updated_callback,
     ADUC_PnPComponentClient_PropertyUpdate_Context* property_update_context)
 {
+    IOTHUB_CLIENT_RESULT iothubInitResult;
+
+    if (g_iothub_client_initialized)
+    {
+        Log_Error("Already initialized.");
```
Comment (JeffMill) [inline src/communication_managers/iothub_communication_manager/src/iothub_communication_manager.c:126]: return true here?

Comment (JeffMill) [inline src/communication_managers/iothub_communication_manager/src/iothub_communication_manager.c:126]: not an error -- info?

File: src/communication_managers/iothub_communication_manager/src/iothub_communication_manager.c [hunk 2]
```diff
@@ -148,6 +169,13 @@ void IoTHub_CommunicationManager_Deinit()
     if (g_aduc_client_handle_address != NULL && *g_aduc_client_handle_address != NULL)
```
Comment (JeffMill) [inline src/communication_managers/iothub_communication_manager/src/iothub_communication_manager.c:169]: why is g_aduc_client_handle_address a pointer?

**Review by JeffMill — CHANGES_REQUESTED** [review]

File: src/communication_managers/iothub_communication_manager/src/iothub_communication_manager.c (hunk 1 above)
Comment (Nox-MSFT) [reply src/communication_managers/iothub_communication_manager/src/iothub_communication_manager.c]: Fixed

Comment (Nox-MSFT) [reply src/communication_managers/iothub_communication_manager/src/iothub_communication_manager.c]: Fixed

File: src/communication_managers/iothub_communication_manager/src/iothub_communication_manager.c [hunk 3]
```diff
@@ -106,23 +111,39 @@ static time_t GetTimeSinceEpochInSeconds()
...
+        Log_Error("Already initialized.");
+    }
+
+    // Before invoking ANY IoTHub Device SDK functionality, IoTHub_Init must be invoked.
+    if ((iothubInitResult = IoTHub_Init()) != 0)
+    {
+        Log_Error("IoTHub_Init failed. Error=%d", iothubInitResult);
+        return false;
+    }
+
```
[the remark this answers is not in the document]
Comment (Nox-MSFT) [reply src/communication_managers/iothub_communication_manager/src/iothub_communication_manager.c:136]: Not exactly. Just need to check here whether IoT Hub stuff is init'ed successfully.

**Review by JeffMill — APPROVED** [review]

**Pull request merged**